packages/core/src/account/smartContractAccount.ts

```diff
@@ -40,10 +47,11 @@ export type AccountOp = {
 };
 
 export type GetAccountParameter<
-  TAccount extends SmartContractAccount | undefined =
-    | SmartContractAccount
+  TEntryPointVersion extends EntryPointVersion,
```

Comment:
I think we should set a default value here like we do everywhere. something like `TEntryPointVersion extends EntryPointVersion = EntrypointVersion` and do that everywhere. this lets us add it to the end of the list to be backwards compatible and makes it a bit easier to work with.

then specific impls can also restrict the version by using 0.6 or 0.7 or keeping the default to signal they support both

Reply:
oh for this param in particular too, if it's a default on `SmartContractAccount` then `GetAccountParameter` doesn't even need this generic